test/AST-Quake/vector-ctor-initlist.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// Test for std::vector initializer_list constructor support
+
+// RUN: cudaq-quake %s | cudaq-opt --canonicalize | FileCheck %s
+
+#include "cudaq.h"
+
+__qpu__ void test() {
+  cudaq::qubit q;
+  std::vector<double> angle{M_PI_2, M_PI_4};
```

Comment:
Similar to my above question, is it worth having a test that does this:
```suggestion
  std::vector<double> angle{M_PI_2, M_PI_4, 99};
```

Reply:
I think there the AST will cast the 99 to 99.0